### PR TITLE
Fixing documentation with respect to array indexing

### PR DESCRIPTION
**Pull request by nelimee**
Main issue: #458 

### Summary
The `comma-separated list of integers contained in braces {a,b,c,…}` way of indexing should not be used for arrays are arrays are meant to be only indexed by single integers and sliced by regular (i.e., not arbitrary) indices. 

This was not cleat at all in the documentation. 
This PR tries to fix this shortcoming.

### Details and comments

The current fix is minimal. I feel like the documentation would benefit from a re-writing for this part. I can see two ways:
1. presenting the different methods of indexing (single integer, integer set, slice) first, and then specifying which method is valid for each array-like types.
2. starting by presenting the most restricted type (arrays) and extending them in the next sections by writing something like "ints are considered as arrays of bits" and "registers (quantum or classical) are considered as arrays of (qu)bits with the additional capability of being indexed by integers sets `{a,b,c,…}`.

### CODE REVIEW

**Review by jakelishman — APPROVED** [review]
Thanks for this - I agree that this is what the array spec part meant, and that any implied "discrete-set indexing" was a mistake, and didn't match the intended behaviour.

**Comment by nelimee** [comment]
Just wanting to check: is there anything I should do that is currently blocking the PR, or is it considered pending until someone actually merge it?

**Comment by jakelishman** [comment]
There's nothing you need to do.  Technically this is a change to the specification (though really it's just a bugfix), so it just needs approval from the technical steering committee.

**Review by blakejohnson — APPROVED** [review]
LGTM

**Pull request merged**
